presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -194,6 +194,7 @@
 import static com.facebook.presto.sql.planner.SystemPartitioningHandle.SINGLE_DISTRIBUTION;
```

Comment:
Actually I noticed issue here in the middle of the work. Later on I added this hack of always adding `TRUE_LITERAL` expression to `JoinNode.filter` for OUTER joins if we end up with empty join condition.
This is temporary hack to force using non-compiled JoinProbe implementation.
This is needed until compiled version can support filter functions.

Side effect of this hack is that expression
```
clauses.isEmpty() && !node.getFilter().isPresent() && node.getType() != INNER
```  
is always false for now.

But there are queries in tests which would fail here if the hack was not present.
E.g. the query I added in fixup commit
```sql
SELECT a FROM (VALUES 1, 2) t1(a) RIGHT OUTER JOIN (VALUES 10, 11) t2(b) ON a > 1
```